packages/main/src/components/AnalyticalTable/ColumnHeader/index.tsx

```diff
@@ -114,30 +118,46 @@ export const ColumnHeader: FC<ColumnHeaderProps> = (props: ColumnHeaderProps) =>
   } = props;
 
   const isFiltered = column.filterValue && column.filterValue.length > 0;
-  const desc = column.isSortedDesc;
-
-  const sortingIcon = column.isSorted ? <Icon name={desc ? 'sort-descending' : 'sort-ascending'} /> : null;
+  const sortingIcon = column.isSorted ? (
+    <Icon name={column.isSortedDesc ? 'sort-descending' : 'sort-ascending'} />
+  ) : null;
   const filterIcon = isFiltered ? <Icon name="filter" /> : null;
   const groupingIcon = column.isGrouped ? <Icon name="group-2" /> : null;
 
+  const textStyle = useMemo(() => {
+    let margin = 0;
+
+    if (column.isSorted) margin++;
+    if (column.isGrouped) margin++;
+    if (isFiltered) margin++;
+
+    if (margin === 0) {
+      return {};
+    }
+
+    if (margin > 0) margin += 0.5;
+
+    if (getRTL()) {
+      return {
+        marginLeft: `${margin}rem`
+      };
+    }
+    return {
+      marginRight: `${margin}rem`
+    };
+  }, [column.isSorted, column.isGrouped, isFiltered, column.hAlign]);
+
   const hasPopover = column.canGroupBy || column.canSort || column.canFilter;
   const innerStyle: CSSProperties = useMemo(() => {
     const modifiedStyles: CSSProperties = {
       cursor: hasPopover ? 'pointer' : 'auto'
     };
-    if (column.canResize) {
-      modifiedStyles.maxWidth = `calc(100% - 16px)`;
-    }
     if (dragOver) {
       modifiedStyles.borderLeft = `3px solid ${ThemingParameters.sapSelectedColor}`;
     }
     if (column.id === '__ui5wcr__internal_highlight_column' || column.id === '__ui5wcr__internal_selection_column') {
       modifiedStyles.padding = 0;
     }
-    if (column.hAlign === TextAlign.End) {
-      modifiedStyles.justifyContent = 'flex-end';
-      modifiedStyles.maxWidth = '';
-    }
     return modifiedStyles;
   }, [column.canResize, dragOver, hasPopover]);
```

Comment:
Please remove unnecessary `column.canResize` from dependencies array.
```suggestion
  }, [dragOver, hasPopover]);
```